### PR TITLE
Disable default nVidia build.

### LINKED ISSUE #55: Building VARIORUM_WITH_GPU likely to break most out-of-the-box builds
VARIORUM_WITH_GPU has a dependence on the nvml.h file provided by nVidia's [CUDA Toolkit](https://developer.nvidia.com/cuda-downloads).  I was not able to pull this down using the standard Ubuntu packages, and I'm concerned the default user experience will either require jumping through nVidia's installation hoops or digging through the CMakeList configuration to turn off that option.

I would rather the folks who want the nVidia build do the extra work.

### PR DESCRIPTION
**Pull request by rountree**
The nvml.h dependence may prevent variorum from being built out-of-the-box for most users.

Sets VARIORUM_WITH_NVIDIA to OFF in src/CMakeLists.txt.

Fixes #55.

### CODE REVIEW

**Comment by slabasan** [comment]
I'm wondering if the out-of-the-box build should only be for Intel (we'd also disable VARIORUM_WITH_IBM and VARIORUM_WITH_AMD).

**Comment by rountree** [comment]
@slabasan If those targets don't require any additional dependences I'm agnostic as to what the default should be.  Your call.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -8,7 +8,7 @@ option(BUILD_TESTS          "Build tests"                  ON)
 option(VARIORUM_WITH_INTEL  "Support Intel architectures"  ON)
 option(VARIORUM_WITH_AMD    "Support AMD architectures"    ON)
 option(VARIORUM_WITH_IBM    "Support IBM architectures"    ON)
```
Comment (slabasan) [inline src/CMakeLists.txt]: Let's disable `VARIORUM_WITH_AMD` and `VARIORUM_WITH_IBM` as well. There's no external dependencies like there is for Nvidia platforms, but many of the systems we see this deployed on are Intel-based.

**Comment by slabasan** [comment]
You can add `Fixes #55.` to the end of your commit, automatically closing the issue here on Github.

File: src/CMakeLists.txt (hunk 1 above)
Comment (rountree) [reply src/CMakeLists.txt]: Done!

**Comment by rountree** [comment]
@slabasan Not sure if the "Fixes #55" trick worked.  Take a look and let me know.

**Comment by slabasan** [comment]
Yep, it worked! Putting the `fixes #<issue>` automatically closes the referenced issue when the PR is merged, thanks!

**Pull request merged**